src/common/src/config.rs

```diff
@@ -540,11 +540,8 @@ mod default {
         }
 
         pub fn total_memory_available_bytes() -> usize {
```

Comment:
To avoid uncertainty between different machines in deterministic test, it's better to return a fixed value on madsim:
```suggestion
        #[cfg(madsim)]
        pub fn total_memory_available_bytes() -> usize {
            16 * 1024 * 1024 * 1024
        }
        #[cfg(not(madsim))]
        pub fn total_memory_available_bytes() -> usize {
```